src/main/java/com/coveros/selenified/utilities/ReportOverview.java

```diff
@@ -0,0 +1,297 @@
+        writer.println("                <thead>");
+        writer.print("                      <tr class='" + overallCssClass + "'>");
+        headerCell("Tests");
+        headerCell("Passed");
+        headerCell("Skipped");
+        headerCell("Failed");
+        headerCell("Time (ms)");
+        writer.println("</tr>");
+        writer.println("                </thead>");
+        writer.println("                <tbody>");
+        writer.print("                      <tr class='" + overallCssClass + "'>");
+        cell(integerFormat.format(totalPassedTests + totalSkippedTests + totalFailedTests));
+        cell(integerFormat.format(totalPassedTests));
+        cell(integerFormat.format(totalSkippedTests));
+        cell(integerFormat.format(totalFailedTests));
+        cell(integerFormat.format(totalDuration));
+        writer.println("</tr>");
+        writer.println("                </tbody>");
+        writer.println("            </table>");
+        writer.println("        </div>");
+    }
+
+    /**
+     * Creates a table with each test case run and it's details
+     */
+    @Override
+    protected void writeScenarioDetails() {
+        writer.println("        <div class='container'>");
+        writer.println("            <table class='table sort'>");
+        writer.println("                <thead>");
+        writer.print("                    <tr>");
+        headerCell("Browser");
+        headerCell("Test Suite");
+        headerCell("Test Case");
+        headerCell("Report");
+        headerCell("Status");
+        headerCell("Time (ms)");
+        writer.println("</tr>");
+        writer.println("                </thead>");
+        writer.println("                <tbody>");
+        for (SuiteResult suiteResult : suiteResults) {
+            for (TestResult testResult : suiteResult.getTestResults()) {
+                writeResults(testResult.getFailedConfigurationResults(), "Fail", "danger");
+                writeResults(testResult.getFailedTestResults(), "Fail", "danger");
+                writeResults(testResult.getSkippedConfigurationResults(), "Skip", "warning");
+                writeResults(testResult.getSkippedTestResults(), "Skip", "warning");
+                writeResults(testResult.getPassedTestResults(), "Pass", "success");
+            }
+        }
+        writer.println("                </tbody>");
+        writer.println("            </table>");
+        writer.println("        </div>");
+        writer.println("    </body>");
+    }
+
+    /**
+     * Helper class to writing out each test case data. extracts the classname, browser, and report information
+     *
+     * @param classResults - the class results containing all of the test and method information
+     * @param status       - did this test pass, fail, or was skipped
+     * @param cssClass     - what color do we want this row to be?
+     */
+    private void writeResults(List<ClassResult> classResults, String status, String cssClass) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS3776)